Packet++/header/VrrpLayer.h

```diff
@@ -0,0 +1,545 @@
+
+		/**
+		* Get authentication type description
+		* @param[in] authType The authentication type
+		* @return The priority in this message
+		*/
+		static std::string getAuthTypeDescByType(uint8_t authType);
+
+		/**
+		 * Get a pointer to the raw VRRPv2/VRRPv3 header. Notice this points directly to the data, so every change will change the actual packet data
+		 * @return A pointer to the @ref vrrp_packet
+		 */
+		vrrp_packet *getPacketPtr() const { return (vrrp_packet *) m_Data; }
+
+		/**
+		 * A method that gets length of virtual address
+		 * @return Virtual IP address length
+		 */
+		uint8_t getIPAddressLen() const;
+
+		/**
+		 * A method that gets VRRP version
+		 * @return VRRP version in this message.
+		 * VRRP version
+		 */
+		uint8_t getVersionFromData() const;
+
+		/**
+		* A method that gets VRRP version
+		* @return The version in this message
+		*/
+		uint8_t getVersion() const;
+
+		/**
+		 * A method that gets VRRP type
+		 * @return VRRP type set in vrrp_packet#type as ::VrrpType enum.
+		 */
+		VrrpType getType() const;
+
+		/**
+		* A method that gets VRRP vrId
+		* @return The virtual router id in this message
+		*/
+		uint8_t getVrId() const;
+
+		/**
+		* A method that gets VRRP priority
+		* @return The priority in this message
+		*/
+		uint8_t getPriority() const;
+
+		/**
+		* A method that gets (maximum) VRRP advertisement interval
+		* @return The (maximum) advertisement interval in this message
+		*/
+		virtual uint16_t getAdvInt() const = 0;
+
+		/**
+		* A method that gets VRRP checksum
+		* @return The checksum in this message
+		*/
+		uint16_t getChecksum() const;
+
+		/**
+		 * A method that gets count of VRRP virtual IP addresses
+		 * @return The count of virtual IP addresses in this message
+		 */
+		uint16_t getIPAddressesCount() const;
+
+		/**
+		 * A method that gets VRRP virtual IP addresses
+		 * @return The virtual IP addresses in this message
+		 */
+		std::vector<IPAddress> getIPAddresses() const;
+
+		/**
+		 * A method that gets first VRRP virtual IP address
+		 * @return A pointer to the first virtual IP address or NULL if no virtual IP address exist. Notice the return value is a pointer to the real data,
+		 * so changes in the return value will affect the packet data
+		 */
+		uint8_t *getFirstIPAddress() const;
+
+		/**
+		 * Get the virtual IP address that comes next to a given virtual IP address. If "ipAddress" is NULL then NULL will be returned.
+		 * If "ipAddress" is the last virtual IP address or if it is out of layer bounds NULL will be returned also. Notice the return value is a
+		 * pointer to the real data casted to IPAddress type (as opposed to a copy of the option data). So changes in the return
+		 * value will affect the packet data
+		 * @param[in] ipAddress The virtual IP address to start searching from
+		 * @return The next virtual IP address or NULL if "ipAddress" is NULL, last or out of layer bounds
+		 */
+		uint8_t *getNextIPAddress(uint8_t *ipAddress) const;
```

Comment:
I'm not sure these methods are really needed. And even if they are needed we should make them private and rename them to `getFirstIPAddressPtr()` and `getNextIPAddressPtr()`.

Essentially all they do is check `getIPAddressLen()` which always returns the same value for a specific packet (so we only need to call it once before starting iterating over the addresses), and then they just move the pointer.

I don't think we need specific methods for that.

Reply:
I decided to keep these methods and move them to be private. Fix in https://github.com/seladb/PcapPlusPlus/pull/1118/commits/0233d684d8aff42029b747845745d835aa066beb